assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -402,8 +517,58 @@ export const controls = {
 	),
 };
 
+// eslint-disable-next-line complexity
 export const reducer = createReducer( ( state, { type, payload } ) => {
 	switch ( type ) {
+		case INSERT_NOTIFICATION_INTO_RESOLVED_QUEUE: {
+			const { id } = payload;
+
+			/**
+			 * The notification we want to add to the already-resolved queue.
+			 */
+			const notification = state.notifications?.[ id ];
+
+			if ( notification === undefined ) {
+				global.console.warn(
+					`Could not add notification with ID "${ id }" to queue. Notification "${ id }" is not registered.`
+				);
+
+				break;
+			}
+
+			state.queuedNotifications[ notification.groupID ] =
+				state.queuedNotifications[ notification.groupID ] || [];
+
+			// If the notification is already in the queue, we don't need to add it
+			// again.
+			if (
+				state.queuedNotifications[ notification.groupID ].some(
+					( notificationInQueue ) => notificationInQueue.id === id
+				)
```

Comment:
Should this maybe be part of the filtering we're already doing to check if a notification should be added or not? Perhaps this and the last check could be done in the same function we have for this purpose. It would just need the added current queue state to check against.

Reply:
Actually I just tried and it's not always relevant, plus it's especially awkward to pass state to the check depending on where it's done.

I know this seems a bit fractured but I think it's more appropriate 🤔